test/unit/plugins/rest_spec.js

```diff
@@ -134,5 +139,237 @@ describe("The Rest plugin", function () {
 				expect(result.statusCode).to.equal(500);
 			});
 		});
+		describe("with an invalid instance", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances/" + INVALID_QUERY).inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("shows the instance does not exist", function () {
+				expect(result.statusCode,"status").to.equal(404);
+			});
+		});
+
+		describe("with a valid instance", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances/" + VALID_INSTANCE_ID).inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("shows the instance", function () {
+				expect(result.statusCode,"status").to.equal(200);
+			});
+		});
+
+		describe("with a valid type", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances?type=" + VALID_TYPE).inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("returns an array of instances with the given type", function () {
+				var payload;
+				expect(result.statusCode,"status").to.equal(200);
+				payload = JSON.parse(result.payload);
+				expect(payload.instances).to.have.length.of.at.least(1);
+			});
+		});
+
+		describe("with an invalid instance by sending non-exisitng Instance Id and type", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances?type=" + INVALID_QUERY + "&id=" + INVALID_QUERY)
+				.inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("returns an empty array of instances", function () {
+				var payload;
+				expect(result.statusCode,"status").to.equal(200);
+				payload = JSON.parse(result.payload);
+				expect(payload.instances.length).equal(0);
+			});
+		});
+
+		describe("with all invalid parameters for a GET Request", function () {
+			var result;
+
+			before(function () {
+				return new Request("GET", "/v1/instances?extra=test&random=thisiswrong")
+				.inject(server)
+				.then(function (response) {
+					result = response;
+				});
+			});
+
+			it("returns a Bad Request", function () {
+				expect(result.statusCode,"status").to.equal(400);
+			});
+		});
+
+		describe("with a mix of valid and invalid parameters", function () {
```

Comment:
Should this case result in a 400? I'm not sure we care if the user adds some extraneous parameters. In fact, there are certain cases where browsers or other middleware will add their own parameters (for instance, `nocache=true`) to a request for various reasons. Doing so shouldn't cause the request to fail if there are valid parameters present.